packages/gotrue/lib/src/types/user_attributes.dart

```diff
@@ -35,6 +38,26 @@ class UserAttributes {
       if (data != null) 'data': data,
     };
   }
+
+  @override
+  bool operator ==(covariant UserAttributes other) {
+    if (identical(this, other)) return true;
+
+    return other.email == email &&
+        other.phone == phone &&
+        other.password == password &&
+        other.nonce == nonce &&
+        other.data == data;
```

Comment:
I think `data` is very often a `Map`, which wouldn't be equal to another Map instance. So I would just use `DeepCollectionEquality()` here instead.